File: backend-dotnet/src/Core/App.Template.Application/DTOs/AuditLogDto.cs
namespace AppTemplate.Application.DTOs;

public class AuditLogDto
{
    public long Id { get; set; }
    public string EntityName { get; set; } = string.Empty;
    public string EntityId { get; set; } = string.Empty;
    public string Action { get; set; } = string.Empty;
    public string? OldValues { get; set; }
    public string? NewValues { get; set; }
    public string? AffectedColumns { get; set; }
    public string? UserId { get; set; }
    public DateTime Timestamp { get; set; }
}

File: backend/spring/clean-architecture/full/application/src/main/java/apptemplate/application/dto/audit/AuditLogDto.java
package apptemplate.application.dto.audit;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuditLogDto {
    private Long id;
    private String entityName;
    private String entityId;
    private String action;
    private String oldValues;
    private String newValues;
    private String affectedColumns;
    private Long userId;
    private LocalDateTime timestamp;
}
